src/components/RecommendationPage.js

```diff
@@ -0,0 +1,27 @@
+import React, { useState, useEffect } from 'react';
+
+const RecommendationPage = ({ userId }) => {
+  const [recommendations, setRecommendations] = useState([]);
+
+  useEffect(() => {
+    fetch(`/recommendations/${userId}`)
```

Comment:
Right now, this is trying to make a call to the front-end itself at a path like: `http://localhost:3000/recommendations/undefined`:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/015b690a-074a-4d55-9d1a-4b7705fd12b8)

That's because you haven't passed in any information to `fetch` for it to know how to call the backend. You could pass in the whole URL you want to call : `http://localhost:8000/api/v1/recommendations` to the fetch call. But then you also need to make sure that when deployed, it calls [`https://musicstore.onrender.com/api/v1/recommendations`](https://musicstore.onrender.com/api/v1/recommendations) instead.  To help with not having to remember to do all this, we already have `axios` client set up in the `apis` folder to handle all this and also handle sending the cookies if the user is already logged in. 

Please update this to use the base client instead. Andwe'll want to also make sure we don't make the API call, if the user id is null (user is not logged in, or parent component has not yet passed in the value).

Reply:
axios implamented , commit 3676f7a3ef5cc36ad31c1b9d1498cd46dbd188d3